luabind/scope.hpp

```diff
@@ -59,13 +59,17 @@ namespace luabind {
     struct LUABIND_API scope
     {
         scope();
+#ifdef LUABIND_USE_CXX11
+        explicit scope(std::unique_ptr<detail::registration> reg);
+#else
         explicit scope(std::auto_ptr<detail::registration> reg);
+#endif
         scope(scope const& other_);
         ~scope();
 
         scope& operator=(scope const& other_);
 
-        scope& operator,(scope s);
+        scope& operator,(const scope& s);
```

Comment:
Reading the implementation in scope.cpp of this, I'm concerned. Was a simple `def("foo", &foo), def("bar", &bar)` previously causing a double-delete?

Would `scope&` without const work here, similar to https://en.cppreference.com/w/cpp/memory/auto_ptr/auto_ptr ?

Reply:
I naively tried `scope&` and the unit tests blew up on me.   It looks like `scope` is trying to implement its own homebrew `unique_ptr` mechanism and that is all tied up in using `const scope&` to trigger the complemented copy constructors to orchestrate the proper ownership transfer.

It felt like fixing this for c++03 would require some serious plumbing work.  However, under c++11, implementing the proper move semantics seems to address the major use cases.